core/server/master/src/main/java/alluxio/master/file/InodeSyncStream.java

```diff
@@ -639,10 +639,8 @@ private void loadMetadata(LockedInodePath inodePath, LoadMetadataContext context
       FileDoesNotExistException, InvalidFileSizeException, InvalidPathException, IOException {
     AlluxioURI path = inodePath.getUri();
     MountTable.Resolution resolution = mMountTable.resolve(path);
-    AlluxioURI ufsUri = resolution.getUri();
-    try (CloseableResource<UnderFileSystem> ufsResource = resolution.acquireUfsResource()) {
-      UnderFileSystem ufs = ufsResource.get();
-      if (context.getUfsStatus() == null && !ufs.exists(ufsUri.toString())) {
+    try {
+      if (context.getUfsStatus() == null) {
```

Comment:
This means for load metadata, this `context.getUfsStatus()` must always be prepopulated with the UFS status? Something earlier in the call stack will be populating everything? What is the difference between this `context.getUfsStatus()` and the status cache?

This also implies this method itself will never call UFS?

Can we add this to the method javadoc?

Reply:
`context.getUfsStatus` is can be thought of as similar to an RPC options object. The `context` object is basically a POJO that just holds information. No fancy processing. Just getters and setters.

When calling this method it is now expected that the ufs status is already populated. And yes, this method shouldn't call the UFS except in the case where the child statuses may need to be retrieved (see the `fetchChildren` call lower in the code)